aws/resource_aws_lightsail_container_service.go

```diff
@@ -0,0 +1,995 @@
+			return diag.Errorf("Lightsail Container Service deployment failed")
+		}
+
+		// disable the container service if needed
+		if newIsDisabled {
+			updateContainerServiceInput := lightsail.UpdateContainerServiceInput{
+				ServiceName: serviceName,
+				IsDisabled:  aws.Bool(true),
+			}
+
+			if _, err := conn.UpdateContainerService(&updateContainerServiceInput); err != nil {
+				log.Printf("[ERROR] Lightsail Container Service (%s) update failed: %s", d.Id(), err)
+				return diag.FromErr(err)
+			}
+
+			stateChangeConf := &resource.StateChangeConf{
+				Pending:    []string{lightsail.ContainerServiceStateUpdating},
+				Target:     []string{lightsail.ContainerServiceStateDisabled},
+				Refresh:    lightsailContainerServiceRefreshFunc(serviceName, meta),
+				Timeout:    25 * time.Minute,
+				Delay:      5 * time.Second,
+				MinTimeout: 3 * time.Second,
+			}
+
+			if _, err := stateChangeConf.WaitForStateContext(ctx); err != nil {
+				log.Printf("[ERROR] Lightsail Container Service (%s) error waiting for container service to disable: %s", d.Id(), err)
+				return diag.FromErr(err)
+			}
+
+			log.Printf("[INFO] Lightsail Container Service (%s) disable successful", d.Id())
+		}
+	} else {
+		// fields handled by UpdateContainerService, can update no matter container service disabled or not
+		publicDomainNames, publicDomainNamesChanged := lightsailContainerServicePublicDomainNamesChanged(d)
+		if d.HasChanges("is_disabled", "power", "scale") || publicDomainNamesChanged {
+			// first determine Target, which is used by StateChangeConf later
+			var targetState string
+			if newIsDisabled {
+				targetState = lightsail.ContainerServiceStateDisabled
+			} else {
+				resp, err := conn.GetContainerServices(&lightsail.GetContainerServicesInput{
+					ServiceName: serviceName,
+				})
+
+				if err != nil {
+					log.Printf("[ERROR] Lightsail Container Service (%s) update failed: %s", d.Id(), err)
+					return diag.FromErr(err)
+				}
+
+				// index 0 because we only queried for 1 container service
+				container := resp.ContainerServices[0]
+
+				if container.CurrentDeployment != nil {
+					targetState = lightsail.ContainerServiceStateRunning
+				} else {
+					targetState = lightsail.ContainerServiceStateReady
+				}
+			}
+
+			updateContainerServiceInput := lightsail.UpdateContainerServiceInput{
+				ServiceName:       serviceName,
+				IsDisabled:        aws.Bool(newIsDisabled),
+				Power:             aws.String(d.Get("power").(string)),
+				PublicDomainNames: publicDomainNames,
+				Scale:             aws.Int64(int64(d.Get("scale").(int))),
+			}
+
+			if _, err := conn.UpdateContainerService(&updateContainerServiceInput); err != nil {
+				log.Printf("[ERROR] Lightsail Container Service (%s) update failed: %s", d.Id(), err)
+				return diag.FromErr(err)
+			}
+
+			stateChangeConf := &resource.StateChangeConf{
```

Comment:
Hi Brittan, thank you for being a Lightsail fan! I also like Lightsail very much!  😎
For the `StateChangeConf`, the reason I rewrite them is that Lightsail container service does not return `Operation` like some other Lightsail resource, and Lightsail container service has relatively more complicated `Pending` and `Target` (there are 7 `StateChangeConf` in total and 6 of them are different). My thinking is that it might be better to leave them there and provide some context in the comments. But the `StateRefreshFunc` can definitely be moved to `aws/internal/service/lightsail/waiter/status.go` after the merge of your PR.

Reply:
I agree that the logic should be moved to the `aws/internal/service/lightsail/waiter/status.go`.

Check out how we did the multiple pending states and single target state in StorageGateway:
https://github.com/hashicorp/terraform-provider-aws/blob/main/aws/internal/service/storagegateway/waiter/waiter.go

Reply:
I support @KaguraKagura's plan

Reply:
Makes sense @KaguraKagura. I am glad we have another lightsail fan working on adding functionality to the provider =) . Hopefully with this PR getting so much attention, some other Lightsail PRs that have been sitting for a while will get some attention as well.